eng/targets/GenerateServiceHubConfigurationFiles.targets

```diff
@@ -0,0 +1,48 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->
+<Project>
+  <!--
+    The inclusion of this file will cause *.servicehub.service.json files to be created. and these json files contains configuration for the servicehub on how it
+    should configure our services in servicehub process
+  -->
+    <Target Name="CalculateServiceHubConfigurationFiles"
+            DependsOnTargets="GetAssemblyVersion">
+    <ItemGroup>
+      <_ServicesWithBitness Include="@(ServiceHubService)" FileSuffix="" HostSuffix=".x86" HostIdSuffix="32" />
+      <_ServicesWithBitness Include="@(ServiceHubService)" FileSuffix="64" HostSuffix="" HostIdSuffix="" />
+
+      <_JsonFile Include="$(IntermediateOutputPath)%(_ServicesWithBitness.FileName)%(_ServicesWithBitness.FileSuffix).servicehub.service.json">
+        <Content>
+          <![CDATA[{
+  "host": "desktopClr%(_ServicesWithBitness.HostSuffix)",
+  "hostId": "RoslynCodeAnalysisService%(_ServicesWithBitness.HostIdSuffix)",
+  "hostGroupAllowed": true,
+  "serviceOverride": true, 
+  "entryPoint": {
+    "assemblyPath": "Microsoft.CodeAnalysis.Remote.ServiceHub.dll",
+    "assemblyName": "Microsoft.CodeAnalysis.Remote.ServiceHub, Version=\"$(AssemblyVersion)\", Culture=neutral, PublicKeyToken=\"$(PublicKeyToken)\"",
+    "fullClassName": "%(_ServicesWithBitness.ClassName)",
+    "appBasePath": "%VSAPPIDDIR%",
+    "configPath": "%PkgDefApplicationConfigFile%"
+  }
+}
+]]>
+        </Content>
+      </_JsonFile>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="GenerateServiceHubConfigurationFiles"
+          DependsOnTargets="CalculateServiceHubConfigurationFiles"
+          AfterTargets="GetVsixSourceItems"
```

Comment:
>GetVsixSourceItems [](start = 24, length = 18)

I don't think this is correct. The target needs to run before `CreateVsixContainer` target runs. Looking at Microsoft.VsSDK.targets, it lokos like the best way to plug into the pipeline is via GetVsixSourceItemsDependsOn:

```xml
<PropertyGroup>
  <GetVsixSourceItemsDependsOn>$(GetVsixSourceItemsDependsOn);GenerateServiceHubConfigurationFiles</GetVsixSourceItemsDependsOn>
</PropertyGroup>
```

Reply:
okay. thank you.